cdk/src/open-data-platform/data-plane/schema/schema.sql

```diff
@@ -31,61 +31,66 @@ $$ LANGUAGE plpgsql;
 -- Census-block-level data. TODO: consider renaming the table.
 
 CREATE TABLE IF NOT EXISTS demographics(
-    census_geo_id varchar(255) NOT NULL,
-    census_block_name varchar(255),
-    total_population real,
+    census_geo_id         varchar(255) NOT NULL,
+    census_block_name     varchar(255),
+    total_population      real,
     under_five_population real,
-    poverty_population real,
-    black_population real,
-    white_population real,
-    geom GEOMETRY(Geometry, 4326),
-    PRIMARY KEY(census_geo_id)
+    poverty_population    real,
+    black_population      real,
+    white_population      real,
+    geom                  GEOMETRY(Geometry, 4326),
+    PRIMARY KEY (census_geo_id)
 );
 
 CREATE INDEX IF NOT EXISTS geom_index
     ON demographics
-    USING GIST (geom);
+        USING GIST (geom);
 
 -- Water-system-level data
 
-CREATE TABLE IF NOT EXISTS water_systems(
-    pws_id varchar(255) NOT NULL,
-    lead_connections_count real,
-    geom GEOMETRY(Geometry, 4326),
-    PRIMARY KEY(pws_id)
-    );
-
-ALTER TABLE water_systems
-    ADD COLUMN IF NOT EXISTS pws_name varchar(255),
-    ADD COLUMN IF NOT EXISTS service_connections_count real,
-    ADD COLUMN IF NOT EXISTS population_served real;
+CREATE TABLE IF NOT EXISTS water_systems
+(
+    pws_id                    varchar(255) NOT NULL,
+    lead_connections_count    real,
+    geom                      GEOMETRY(Geometry, 4326),
+    pws_name                  varchar(255),
+    service_connections_count real,
+    population_served         real,
+    state_census_geo_id              varchar(255),
+    county_census_geo_id             varchar(255),
+    PRIMARY KEY (pws_id)
+);
 
 CREATE INDEX IF NOT EXISTS geom_index
     ON water_systems
-    USING GIST (geom);
+        USING GIST (geom);
 
 -- EPA violations data
 
-CREATE TABLE IF NOT EXISTS epa_violations(
-    violation_id varchar(255) NOT NULL,
-    violation_code varchar(255) NOT NULL,
+CREATE TABLE IF NOT EXISTS epa_violations
+(
+    violation_id      varchar(255) NOT NULL,
+    violation_code    varchar(255) NOT NULL,
     compliance_status varchar(255) NOT NULL,
-    start_date DATE NOT NULL,
-    end_date DATE,
-    pws_id varchar(255) NOT NULL,
-    PRIMARY KEY(violation_id)
-    );
+    start_date        DATE         NOT NULL,
+    end_date          DATE,
+    pws_id            varchar(255) NOT NULL,
+    state_census_geo_id      varchar(255),
+    county_census_geo_id     varchar(255),
+    PRIMARY KEY (violation_id)
+);
 
 -- Violation counts per water system --
 
 CREATE OR REPLACE VIEW violation_counts AS
 SELECT
     pws_id,
     geom,
-    COUNT(violation_id) AS violation_count
+    COUNT(violation_id) AS violation_count,
+    epa_violations.state_census_geo_id
 FROM epa_violations
 JOIN water_systems USING (pws_id)
-GROUP BY pws_id, geom;
+GROUP BY pws_id, geom, epa_violations.state_census_geo_id;
```

Comment:
It might be a good idea to make an index for `state_census_geo_id` since we're grouping by it.

Reply:
Done.